core/templates/dev/head/services/PlaythroughIssuesService.ts

```diff
@@ -183,9 +186,42 @@ oppia.factory('PlaythroughIssuesService', [
                     displayBlock, blockActionIndexMapping[index]);
                 };
 
+                /**
+                 * Returns the index of the learner action wihtin the display
+                 * block.
+                 * @param {LearnerAction} learnerAction.
+                 * @param {LearnerAction[]} displayBlock.
+                 * @returns {int}
+                 */
+                $scope.getLearnerActionIndex = function(
+                    learnerAction, displayBlock) {
+                  return displayBlock.indexOf(learnerAction);
+                };
+
+                /**
+                 * Renders the HTML of the learner action. The index of the
+                 * learner action will be the sum of the starting action index
+                 * of the block (a block is a list of learner actions grouped
+                 * together for display) computed using the
+                 * blockActionIndexMapping field and the index of the learner
+                 * action within the block.
+                 * @param {LearnerAction} learnerAction.
+                 * @param {int} blockIndex - The index of the block among all
+                 *  the display blocks.
+                 * @param {int} actionIndex - The index of the learner action
+                 *  within it's display block.
+                 * @returns {string}
+                 */
+                $scope.renderLearnerAction = function(
+                    learnerAction, blockIndex, actionIndex) {
+                  return LearnerActionRenderService.renderLearnerAction(
+                    learnerAction, blockActionIndexMapping[blockIndex],
+                    actionIndex);
+                };
+
                 var getRemainingActionsElements = function(pIdx, i) {
-                  return document.getElementById(
-                    'remainingActions' + pIdx.toString() + i.toString());
+                  return document.getElementsByClassName(
```

Comment:
Add a line comment above this line saying we only expect one element to match (to justify the [0]).

Reply:
Done.